components/packages/index.rst

```diff
@@ -0,0 +1,259 @@
+      - <<: !include common/sensor/wifi_signal.config.yaml
+    binary_sensor:
+      - <<: !include common/binary_sensor/connection_status.config.yaml
+
+    switch:
+      - <<: !include common/switch/restart_switch.config.yaml
+
+.. _config-git_packages:
+
+Remote/Git Packages
+-------------------
+
+Packages can also be loaded from a Git repository by utilizing the correct configuration syntax.
+:doc:`/components/substitutions/index` can be used inside the remote packages which allows users to override
+them locally with their own substitution value.
+
+.. note::
+
+    Remote packages cannot have ``secret`` lookups in them. They should instead make use of substitutions with an
+    optional default in the packaged YAML, which the local device YAML can set using values from the local secrets.
+
+.. code-block:: yaml
+
+    packages:
+      # Git repo examples
+      remote_package:
+        url: https://github.com/esphome/non-existant-repo
+        ref: main # optional
+        files: [file1.yml, file2.yml]
+        refresh: 1d # optional
+
+      # A single file can be expressed using `file` or `files` as a string
+      remote_package_two:
+        url: https://github.com/esphome/non-existant-repo
+        file: file1.yml # cannot be combined with `files`
+        # files: file1.yml
+
+      # shorthand form github://username/repository/[folder/]file-path.yml[@branch-or-tag]
+      remote_package_three: github://esphome/non-existant-repo/file1.yml@main
+
+Packages as Templates
+---------------------
+
+Since packages are incorporated using the ``!include`` system, variables can be provided to them. This means that
+packages can be used as *templates*, allowing complex or repetitive configurations to be stored in a package file
+and then incorporated into the configuration more than once.
+
+Packages may also contain a ``defaults`` block which provides subsitutions for variables not provided by the
+``!include`` block.
+
+As an example, if the configuration needed to support three garage doors using the ``gpio`` switch platform and the
+``time_based`` cover platform, it could be constructed like this:
+
+.. code-block:: yaml
+
+    # In config.yaml
+    packages:
+      left_garage_door: !include
+        file: garage-door.yaml
+        vars:
+          door_name: Left
+          door_location: left
+          open_switch_gpio: 25
+          close_switch_gpio: 26
+      middle_garage_door: !include
+        file: garage-door.yaml
+        vars:
+          door_name: Middle
+          door_location: middle
+          open_switch_gpio: 27
+          close_switch_gpio: 29
+      right_garage_door: !include
+        file: garage-door.yaml
+        vars:
+          door_name: Right
+          door_location: right
+          open_switch_gpio: 15
+          close_switch_gpio: 18
+          open_duration: "1min"
+          close_duration: "50s"
+
+
+.. code-block:: yaml
+
+    # In garage-door.yaml
+    defaults:
```

Comment:
I use "substitutions" here what is the difference?

Reply:
the difference is that `defaults` are scoped to the package only. `substitutions` "leak" out to the root yaml merge.